src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1120,6 +1135,78 @@ FinalizeRouterPlan(PlannedStmt *localPlan, CustomScan *customScan)
 }
 
 
+/*
+ * BlessRecordExpression ensures we can parse an anonymous composite type on the
```

Comment:
This function is doing some allocations within the current memory context. That all seems fine because `assign_record_type_typmod` uses the cache context when it needs to store the tupleDesc, and we're not relying on any memory generated via this function.  

Do you see any issues with the memory contexts?

Reply:
Should be fine. Fortunately we're only flipping an integer in the plan so we don't need to allocate new memory.

`get_expr_result_type` explicitly returns a copy of the TupleDesc in the current context.